experimental/Schemes/AffineCurve.jl

```diff
@@ -0,0 +1,42 @@
+export AffineCurve
+export underlying_scheme
+export ambient_space
+
+mutable struct AffineCurve{BRT, BRET, SpecType<:Spec, PolyType<:MPolyElem} <: AbsSpec{BRT, BRET}
+  C::SpecType # the underlying scheme
+  A::SpecType # the ambient space
+  f::Vector{PolyType} # the defining equations
+
+  # Construct the affine curve in Spec(R) which is the zero locus of fⱼ in the hypersurface 
+  # complement of Πᵢ gᵢ.
+  function AffineCurve(R::MPolyRing, f::Vector{PolyType}, g::Vector{PolyType}) where {PolyType<:MPolyElem}
+    all(x->(parent(x)==R), f) || error("polynomials do not belong to the correct ring")
+    all(x->(parent(x)==R), g) || error("polynomials do not belong to the correct ring")
+    A = Spec(R)
+    I = ideal(R, f)
+    U = MPolyPowersOfElement(R, g)
+    L = MPolyQuoLocalizedRing(R, I, U)
+    C = Spec(L)
+    dim(C) == 1 || error("the given generators do not define a subscheme of dimension 1")
+    return new{typeof(coefficient_ring(R)), elem_type(coefficient_ring(R)), typeof(C), PolyType}(C, A, f)
+  end
+end
+
+### overwriting the required getters
+function OO(C::AffineCurve) 
+  return OO(C.C)
+end
+
+### overwriting the required type-getters
+ring_type(::Type{CurveType}) where {BRT, BRET, SpecType, PolyType, CurveType<:AffineCurve{BRT, BRET, SpecType, PolyType}} = ring_type(SpecType)
+
+### additional getters
+underlying_scheme(C::AffineCurve) = C.C
```

Comment:
Would just `scheme` work?

Reply:
Yes, that would work equally well. Keep in mind that this is just a rough sample implementation; the particular names are not that important. I just wanted to give an example for another getter, that is not standard, but reasonable.

Reply:
Ah no, it should not be `scheme`, because `C isa Scheme`, so `scheme(C)` does not make much sense. This is just an internal function, right?